api/src/test/java/com/epam/pipeline/controller/contextual/ContextualPreferenceControllerTest.java

```diff
@@ -0,0 +1,219 @@
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = ContextualPreferenceController.class)
+public class ContextualPreferenceControllerTest extends AbstractControllerTest {
+
+    private static final String TEST_STRING = "TEST";
+    private static final String CONTEXTUAL_URL = SERVLET_PATH + "/contextual/preference";
+    private static final String LOAD_URL = CONTEXTUAL_URL + "/load";
+    private static final String LOAD_ALL_URL = LOAD_URL + "/all";
+    private static final ContextualPreferenceLevel PREFERENCE_LEVEL = ContextualPreferenceLevel.ROLE;
+    private ContextualPreference contextualPreference;
+    private ResponseResult<ContextualPreference> expectedResult;
+    private ContextualPreferenceExternalResource contextualPreferenceExternalResource;
+
+    @Autowired
+    private ContextualPreferenceApiService mockContextualPreferenceApiService;
+
+    @Before
+    public void setUp() {
+        contextualPreference = ContextualPreferenceCreatorUtils.getContextualPreference();
+        expectedResult = ControllerTestUtils.buildExpectedResult(contextualPreference);
+
+        contextualPreferenceExternalResource
+                = ContextualPreferenceCreatorUtils.getCPExternalResource();
+    }
+
+    @Test
+    public void shouldFailLoadAllForUnauthorizedUser() throws Exception {
+        mvc().perform(get(LOAD_ALL_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAll() throws Exception {
+        final List<ContextualPreference> contextualPreferences = Collections.singletonList(contextualPreference);
+
+        Mockito.doReturn(contextualPreferences).when(mockContextualPreferenceApiService).loadAll();
+
+        final MvcResult mvcResult = mvc().perform(get(LOAD_ALL_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        Mockito.verify(mockContextualPreferenceApiService).loadAll();
+
+        final ResponseResult<List<ContextualPreference>> expectedResult =
+                ControllerTestUtils.buildExpectedResult(contextualPreferences);
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult,
+                new TypeReference<Result<List<ContextualPreference>>>() { });
```

Comment:
Probably we can move this verbose type reference instantiation to some other place like `ContextualPreferenceCreatorUtils`. It can be pretty handy to use it later on like the following

```java
ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult, 
        ContextualPreferenceCreatorUtils.contextualPreferencesListType);
// or
ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult, 
        ContextualPreferenceCreatorUtils.listOfContextualPreferencesType);
```